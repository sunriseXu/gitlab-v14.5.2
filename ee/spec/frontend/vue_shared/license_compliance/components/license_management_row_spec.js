import { shallowMount } from '@vue/test-utils';

import LicenseManagementRow from 'ee/vue_shared/license_compliance/components/license_management_row.vue';
import {
  approvedLicense,
  blacklistedLicense,
} from 'ee_jest/vue_shared/license_compliance/mock_data';

let wrapper;

describe('LicenseManagementRow', () => {
  afterEach(() => {
    wrapper.destroy();
  });

  describe('allowed license', () => {
    beforeEach(() => {
      const props = { license: approvedLicense };

      wrapper = shallowMount(LicenseManagementRow, {
        propsData: {
          ...props,
        },
      });
    });

    it('renders the license name', () => {
      expect(wrapper.find('.name').element).toMatchSnapshot();
    });

    it('renders the allowed status text with the status icon', () => {
      expect(wrapper.find('.status').element).toMatchSnapshot();
    });
  });

  describe('denied license', () => {
    beforeEach(() => {
      const props = { license: blacklistedLicense };

      wrapper = shallowMount(LicenseManagementRow, {
        propsData: {
          ...props,
        },
      });
    });

    it('renders the license name', () => {
      expect(wrapper.find('.name').element).toMatchSnapshot();
    });

    it('renders the denied status text with the status icon', () => {
      expect(wrapper.find('.status').element).toMatchSnapshot();
    });
  });
});
