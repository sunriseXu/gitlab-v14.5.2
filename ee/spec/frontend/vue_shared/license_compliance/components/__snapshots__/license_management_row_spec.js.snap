// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseManagementRow allowed license renders the allowed status text with the status icon 1`] = `
<div
  class="table-mobile-content text-capitalize d-flex align-items-center justify-content-end justify-content-md-start status"
>
  <issue-status-icon-stub
    status="success"
    statusiconsize="24"
  />
  
        Allowed
      
</div>
`;

exports[`LicenseManagementRow allowed license renders the license name 1`] = `
<div
  class="table-mobile-content name"
>
  
        MIT
      
</div>
`;

exports[`LicenseManagementRow denied license renders the denied status text with the status icon 1`] = `
<div
  class="table-mobile-content text-capitalize d-flex align-items-center justify-content-end justify-content-md-start status"
>
  <issue-status-icon-stub
    status="failed"
    statusiconsize="24"
  />
  
        Denied
      
</div>
`;

exports[`LicenseManagementRow denied license renders the license name 1`] = `
<div
  class="table-mobile-content name"
>
  
        New BSD
      
</div>
`;
